edb/edgeql/compiler/stmt.py

```diff
@@ -75,12 +75,13 @@ def compile_SelectQuery(
             sctx.partial_path_prefix = ctx.partial_path_prefix
             stmt.implicit_wrapper = True
 
-        if expr.limit is not None:
-            sctx.inhibit_implicit_limit = True
-        elif ((ctx.expr_exposed or sctx.stmt is ctx.toplevel_stmt)
-                and ctx.implicit_limit and not sctx.inhibit_implicit_limit):
+        if (
+            (ctx.expr_exposed or sctx.stmt is ctx.toplevel_stmt)
+            and ctx.implicit_limit
+            and expr.limit is None
+            and not ctx.inhibit_implicit_limit
+        ):
```

Comment:
(What's the change happening here?)

Reply:
The previous coding attempted to solve this case:

```
SELECT
   (SELECT ... LIMIT <implicit>) # nested
LIMIT
   100 # more than implicit
```

But this is stupid and unnecessary, because the CLI would trim excessive entries anyway, and so the new approach doesn't care if your explicit `LIMIT` is greater than implicit, it'll clamp to the least of the two.